### PR TITLE
Initial changes to asm module to support installing asm ver 1.8.

### PR DESCRIPTION
**Pull request by kaariger**
Modified default value for asm_version to 1.8
Added additional IAM bindings and API enablement to the test scripts
Modified install_asm.sh wrapper script to call the new install_asm script with the updated flags.
- Need to remove the --enable_cluster_labels flag in the wrapper script once the install_asm script makes that flag optional. It is currently required.

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_generate_modules
 The modules need to be regenerated. Please run `make_build`. 
 ```bash 
Checking submodule's files generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' /workspace/modules/asm/README.md /tmp/tmp.fZXVS9Xetr/workspace/modules/asm/README.md
41c41
< | asm\_version | ASM version to deploy. Available versions are documented in https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages | `string` | `"release-1.6-asm"` | no |
---
> | asm\_version | ASM version to deploy. Available versions are documented in https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages | `string` | `"1.8"` | no |
Error: submodule's files generation has not been run, please run the
'make build' command and commit changes
 ``` 
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/modules/asm/README.md /tmp/tmp.LeZoUrNLnn/generate_docs/workspace/modules/asm/README.md
41c41
< | asm\_version | ASM version to deploy. Available versions are documented in https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages | `string` | `"release-1.6-asm"` | no |
---
> | asm\_version | ASM version to deploy. Available versions are documented in https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages | `string` | `"1.8"` | no |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ``` 
- ⚠️check_shell
 Failed shell check. More info on running shellcheck locally [here](https://www.shellcheck.net). 
 ```bash 
Running shellcheck
In ./modules/asm/scripts/install_asm.sh line 43:
./install_asm --verbose --project_id ${PROJECT_ID} --cluster_name ${CLUSTER_NAME} --cluster_location ${CLUSTER_LOCATION} --mode ${MODE} --output_dir ${BASE_DIR} --enable_cluster_labels --enable_cluster_roles
                                     ^-----------^ SC2086: Double quote to prevent globbing and word splitting.
                                                                  ^-------------^ SC2086: Double quote to prevent globbing and word splitting.
                                                                                                     ^-----------------^ SC2086: Double quote to prevent globbing and word splitting.
For more information:
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
 ```

File: modules/asm/scripts/install_asm.sh [hunk 1]
```diff
@@ -15,59 +15,29 @@
...
-# fi
-gcloud beta anthos apply ${BASE_DIR}
-kubectl wait --for=condition=available --timeout=600s deployment --all -n istio-system
+
+#download the correct version of the install_asm script
+curl https://storage.googleapis.com/csm-artifacts/asm/install_asm_"${ASM_VERSION}" > install_asm
+chmod u+x install_asm
+
+#run the script with appropriate flags
+./install_asm --verbose --project_id ${PROJECT_ID} --cluster_name ${CLUSTER_NAME} --cluster_location ${CLUSTER_LOCATION} --mode ${MODE} --output_dir ${BASE_DIR} --enable_cluster_labels --enable_cluster_roles
```
Comment (bharathkkb) [inline modules/asm/scripts/install_asm.sh:43]: Is an output dir necessary, IIRC the asm script can create a tmp dir to hydrate the configs?
/cc @zerobfd

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks for the PR @kaariger

File: modules/asm/scripts/install_asm.sh (hunk 1 above)
Comment (zerobfd) [reply modules/asm/scripts/install_asm.sh:43]: Yep, by default it creates a /tmp directory and re-downloads the tarball and kpt pkg every time.

If you specify a stable output directory and it already has the kpt package and binary, it won't re-download anything, so you could use that to pin on a certain version until someone deleted the directory.

**Pull request closed without merging**